src/main/java/com/panxoloto/sharepoint/rest/PLGSharepointOnPremisesClient.java

```diff
@@ -565,7 +578,11 @@ public JSONObject uploadFile(String folder, Resource resource, String fileName,
 		MultiValueMap<String, String> headers = headerHelper.getPostHeaders("");
 	    headers.remove("Content-Length");
 
-	    RequestEntity<Resource> requestEntity = new RequestEntity<>(resource, 
+		byte[] bytes = resource.getContentAsByteArray();
```

Comment:
Need to Change into byte[] because without Content-Length header upload result was corrupted file 
so used **byte[] bytes = resource.getContentAsByteArray();** to get file upload size